pyproject.toml

```diff
@@ -1,10 +1,98 @@
+[project]
+name = "rockcraft"
+dynamic = ["version"]
+description = "Create rocks"
+license = {file = "LICENSE"}
+readme = "README.rst"
+requires-python = ">= 3.10"
+authors = [
+    { name = "Canonical Ltd.", email = "snapcraft@lists.snapcraft.io" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "craft-application>=4.0.0",
+    "craft-archives>=2.0.0",
+    "craft-cli",
+    "craft-parts>=2.1.2",
+    "craft-platforms~=0.3",
+    "craft-providers>=2.0.4",
+    "overrides",
+    "spdx-lookup",
+    "tabulate>=0.8.10",
+]
+
+[project.scripts]
+rockcraft = "rockcraft.cli:run"
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "canonical-sphinx[full]",
+    "codespell",
+    "coverage",
+    "mypy",
+    "pyspelling",
+    "pytest",
+    "pytest-check>=2.0",
+    "pytest-mock",
+    "pytest-subprocess",
+    "sphinx-autobuild",
+    "sphinx-autodoc-typehints",
+    "sphinx-lint",
+    "sphinx-toolbox",
+    "sphinxcontrib-details-directive",
+    "twine",
+    "types-pyyaml",
+    "types-requests",
+    "types-setuptools",
+    "types-tabulate>=0.9.0.2",
+    "wheel",
+]
+doc = [
+    "canonical-sphinx[full]",
+    "pyspelling",
+    "sphinx-autobuild",
+    "sphinx-autodoc-typehints",
+    "sphinx-lint",
+    "sphinx-toolbox",
+    "sphinxcontrib-details-directive",
+]
+release = [
+    "twine",
+    "wheel",
+]
+
+[project.urls]
+Documentation = "https://rockcraft.readthedocs.io/en/latest/"
+Homepage = "https://github.com/canonical/rockcraft"
+Issues = "https://github.com/canonical/rockcraft/issues"
+Source = "https://github.com/canonical/rockcraft.git"
+
 [build-system]
 requires = [
-    "setuptools~=70.0.0",
+    "setuptools>=69.0",
     "setuptools_scm[toml]>=7.1"
 ]
 build-backend = "setuptools.build_meta"
 
+[tool.setuptools.dynamic]
+readme = {file = "README.rst"}
+
+[tool.setuptools.packages.find]
+include = ["rockcraft"]
+namespaces = false
+
+[tool.setuptools.package-data]
+"share.rockcraft" = ["extensions"]
```

Comment:
@lengau the spread tests are failing because they expect to find the files from `<root>/extensions` at `'/snap/rockcraft/x1/share/rockcraft/extensions/flask-framework'`. I guess this worked before because we were 'abusing' the setup.py setup to add files outside the package dir

Am I correct in thinking we should move away from that? We'll need to preserve that path on the snap for backwards-comp but maybe those data files should go in `rockcraft/extensions/data` (or similar)

Reply:
I updated this by adding `extensions` as a separate package and then moving it from site-packages to share/rockcraft/ when packing the snap